### PR TITLE
Patch DBs in django app config

### LINKED ISSUE #1014: Measuring Django management commands
When I use `patch_all(django=True)` inside my management command it doesn't behave the same as the API.

For context, I have a management command that does this:
1. Calls `patch_all`
1. Gets a message off the queue from SQS
1. calls `tracer.context_provider.activate(context)` on the context container in the message
1. creates a span with some tags

What I'm finding, is that the ORM/cache instrumentation doesn't seem to be working.
However, if I try to access `connections._datadog_original_connections_all` it exists which leads me to believe it is patched.

On a side note: we aren't getting an `apdex` score for our django applications even though they are correctly detected as `web` and have traces as such.

### PR DESCRIPTION
**Pull request by JBKahn**
Fixes #1014

Management commands don't call `connections.all()` anywhere by default and as a result, are excluded from being patched unless a user manually makes that call. This will instead call this inside the AppConfig for the Django app so that all processes benefit from tracing by default.

This also moves the code into a function outside the ready which solves 2 things:
1. It's easier to test.
1. It means we can reuse this code in our django 1.6 app rather than copying it in.

### CODE REVIEW

**Comment by JBKahn** [comment]
Unsure why this would break the Mongo one, it seems to want to re-use django based things but I'm still not sure. Any insight from you guys?

**Comment by JBKahn** [comment]
I'll take a look today at the tests and add one in an hour or so

**Comment by JBKahn** [comment]
Hmm right now I haven't thought of a way to test it @brettlangdon since the tests all have `connections.all` run before they are run as I don't see that being trigger anywhere. I'm not sure.

File: ddtrace/contrib/django/patch.py [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+log = get_logger(__name__)
 
 
 def patch():
     """Patch the instrumented methods
     """
     if getattr(django, '_datadog_patch', False):
         return
-    setattr(django, '_datadog_patch', True)
+    setattr(django, '_datadog_pacth', True)
```
Comment (nickpresta) [inline ddtrace/contrib/django/patch.py]: typo:

```suggestion
    setattr(django, '_datadog_patch', True)
```

Comment (JBKahn) [reply ddtrace/contrib/django/patch.py]: had to fix both lines. good catch

**Comment by JBKahn** [comment]
@brettlangdon any suggestions on how to test this one? I thought the mock test would work but I don't think I can mock the `patch.connections` because it's confusing it with the patch function I think.

**Comment by JBKahn** [comment]
rebasing locally did interesting things...

**Comment by JBKahn** [comment]
@brettlangdon this should now be ready for review.

**Comment by nickpresta** [comment]
Ping. Is there anything we can do to help get this merged into `master` (and presumably `0.29.0`)?

**Pull request merged**